starter-code/src/functions-and-arrays.js

```diff
@@ -1,100 +1,442 @@
 // Find the maximum
-
+function maxOfTwoNumbers(num1,num2){
+  if (num1>num2){return num1}
+  else if (num2>num1){return num2}
+  else{return num1,num2}
+}
 // Finding Longest Word
 var words = [
-  'mystery',
-  'brother',
-  'aviator',
-  'crocodile',
-  'pearl',
-  'orchard',
-  'crackpot'
+  "mystery",
+  "brother",
+  "aviator",
+  "crocodile",
+  "pearl",
+  "orchard",
+  "crackpot"
 ];
 
+function findLongestWord(arrayLong) {
+  var longestWord = "";
+
+  for (var i = 0; i <= arrayLong.length - 1; i++) {
+    if (arrayLong[i].length > longestWord.length) {
+      longestWord = arrayLong[i];
+    }
+  }
+  
+  return longestWord;
+}
+
 // Calculating a Sum
 
 var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
 
+function sumArray (arraySum){
+
+ var summedArray = arraySum.reduce(function(a,b){
+    return a+b;
+  })
+ return summedArray;
+}
+
 // Calculate the Average
 
 var numbersAvg = [2, 6, 9, 10, 7, 4, 1, 9];
 
+function averageNumbers (arrayAverage){
+
+ return sumArray(arrayAverage)/arrayAverage.length
+
+}
+
 // Array of Strings
 var wordsArr = [
-  'seat',
-  'correspond',
-  'linen',
-  'motif',
-  'hole',
-  'smell',
-  'smart',
-  'chaos',
-  'fuel',
-  'palace'
+  "seat",
+  "correspond",
+  "linen",
+  "motif",
+  "hole",
+  "smell",
+  "smart",
+  "chaos",
+  "fuel",
+  "palace"
 ];
 
 // Unique Arrays
 var wordsUnique = [
-  'crab',
-  'poison',
-  'contagious',
-  'simple',
-  'bring',
-  'sharp',
-  'playground',
-  'poison',
-  'communion',
-  'simple',
-  'bring'
+  "crab",
+  "poison",
+  "contagious",
+  "simple",
+  "bring",
+  "sharp",
+  "playground",
+  "poison",
+  "communion",
+  "simple",
+  "bring"
 ];
 
 // Finding Elements
 var wordsFind = [
-  'machine',
-  'subset',
-  'trouble',
-  'starting',
-  'matter',
-  'eating',
-  'truth',
-  'disobedience'
+  "machine",
+  "subset",
+  "trouble",
+  "starting",
+  "matter",
+  "eating",
+  "truth",
+  "disobedience"
 ];
 
+  function averageWordLength(arrayAverageWordLength){
+    var arrayWithLengthWords = arrayAverageWordLength.map(function(a){
+      return a.length;
+    })
+    return averageNumbers(arrayWithLengthWords);
+  }
+
 // Counting Repetion
 var wordsCount = [
-  'machine',
-  'matter',
-  'subset',
-  'trouble',
-  'starting',
-  'matter',
-  'eating',
-  'matter',
-  'truth',
-  'disobedience',
-  'matter'
+  "machine",
+  "matter",
+  "subset",
+  "trouble",
+  "starting",
+  "matter",
+  "eating",
+  "matter",
+  "truth",
+  "disobedience",
+  "matter"
 ];
+
+function uniquifyArray (arrayUniquify){
+  var finalArray = [];
+  for (var j=0; j<=arrayUniquify.length;j++){
+    if (arrayUniquify.indexOf(arrayUniquify[j])==j){
```

Comment:
Here more logically you would want to do the indexOf in the finalArray. If the indexOf returns -1 then it does not exist, so push it into the array.